4.queueRange.js

```diff
@@ -35,7 +35,18 @@
 
 // Write your function here
 const range = (queue) => {
-
+    q = queue
+    if(q.isEmpty()) return null;
+    let min = q.peek()
+     let max = q.peek()
+    while(!q.isEmpty()){
+        max = Math.max(max, q.dequeue())
+    }
```

Comment:
At this point our queue would be empty. Instead save the dequeue variable and move the min logic into the same loop.